### PR TITLE
Don't require users to be binded to the LDAP server to do a search

### PR DESCRIPTION
**Pull request by fanta8897**

### CODE REVIEW

**Comment by snipe** [comment]
Notes from Gitter:

> If that user isn't binded on the ldap server, which is a security precaution, they can't do a directory search and as such Auth fails since it's in the same try except

**Comment by fanta8897** [comment]
How would this go about becoming merged?

Thanks

File: app/Models/Ldap.php [hunk 1]
```diff
@@ -105,6 +105,10 @@ static function findAndBindUserLdap($username, $password)
         if (!$ldapbind = @ldap_bind($connection, $userDn, $password)) {
             return false;
         }
+        
+        ldap_unbind($connection);
```
Comment (snipe) [inline app/Models/Ldap.php]: If ldapbind fails, the function returns false, so the unbind never fires, no?

**Comment by snipe** [comment]
Haven't heard back, so I'm going to close this ticket for now, but will re-open it if you're still having issues

File: app/Models/Ldap.php (hunk 1 above)
Comment (fanta8897) [reply app/Models/Ldap.php]: ldap_unbind would only work if there is a successful bind, so if the bind fails, we would not want to run unbind. If it succeeds, we would want to initiate the new connection as the admin user to run the search with the respective filter query

**Pull request closed without merging**